src/main/java/com/iexec/sms/secret/teetaskruntime/AppRuntimeSecretController.java

```diff
@@ -0,0 +1,299 @@
+
+    private static final Map<String, String> invalidAuthorizationPayload = createErrorPayload("Invalid authorization");
+
+    public AppRuntimeSecretController(AuthorizationService authorizationService,
+                                      TeeTaskRuntimeSecretService teeTaskRuntimeSecretService,
+                                      TeeTaskRuntimeSecretCountService teeTaskRuntimeSecretCountService) {
+        this.authorizationService = authorizationService;
+        this.teeTaskRuntimeSecretService = teeTaskRuntimeSecretService;
+        this.teeTaskRuntimeSecretCountService = teeTaskRuntimeSecretCountService;
+    }
+
+    // region App developer endpoints
+    @PostMapping("/apps/{appAddress}/secrets/0")
+    public ResponseEntity<Map<String, String>> addAppDeveloperAppRuntimeSecret(@RequestHeader("Authorization") String authorization,
+                                                                  @PathVariable String appAddress,
+//                                                                      @PathVariable long secretIndex,    // FIXME: enable once functioning has been validated
+                                                                  @RequestBody String secretValue) {
+        long secretIndex = 0;   // FIXME: remove once functioning has been validated.
+
+        if (!SecretUtils.isSecretSizeValid(secretValue)) {
+            return ResponseEntity
+                    .status(HttpStatus.PAYLOAD_TOO_LARGE)
+                    .body(createErrorPayload("Secret size should not exceed 4 Kb"));
+        }
+
+        String challenge = authorizationService.getChallengeForSetAppRuntimeSecret(appAddress, secretIndex, secretValue);
+
+        if (!authorizationService.isSignedByOwner(challenge, authorization, appAddress)) {
+            log.error("Unauthorized to addAppDeveloperAppRuntimeSecret" +
+                            " [appAddress: {}, expectedChallenge: {}]",
+                    appAddress, challenge);
+            return ResponseEntity
+                    .status(HttpStatus.UNAUTHORIZED)
+                    .body(invalidAuthorizationPayload);
+        }
+
+        if (teeTaskRuntimeSecretService.isSecretPresent(
+                OnChainObjectType.APPLICATION,
+                appAddress,
+                SecretOwnerRole.APPLICATION_DEVELOPER,
+                null,
+                secretIndex)) {
+            log.error("Can't add app developer secret as it already exists" +
+                            " [appAddress:{}, secretIndex:{}]",
+                    appAddress, secretIndex);
+            return ResponseEntity
+                    .status(HttpStatus.CONFLICT)
+                    .body(createErrorPayload("Secret already exists"));
+        }
+
+        teeTaskRuntimeSecretService.encryptAndSaveSecret(
+                OnChainObjectType.APPLICATION,
+                appAddress,
+                SecretOwnerRole.APPLICATION_DEVELOPER,
+                null,
```

Comment:
What do you think of putting `address(0)` `(0x00...0`) for secrets with no `ownerAddress`. It makes the information explicit and keeps some coherence. That way we make the difference between secrets without `ownerAddress` and buggy secrets.

Reply:
We just need to make sure it does not introduce any problem first.

Reply:
`""` is fine for me, it's just that we will run into trouble whenever we decide to validate ETH addresses. But this is minor.

Reply:
Alright, that's done! Thanks for your suggestions :)